torch/distributed/internal_rpc_utils.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import pickle
+import copyreg
+import io
+import collections
+
+def serialize(obj):
+    f = io.BytesIO()
+    p = pickle.Pickler(f)
+    if hasattr(p, 'dispatch'):
+        # Make the dispatch registry an instance level attribute instead of
+        # a reference to the class dictionary under Python 2
+        p.dispatch = pickle.Pickler.dispatch.copy()
+    else:
+        # Under Python 3 initialize the dispatch table with a copy of the
+        # default registry
+        p.dispatch_table = copyreg.dispatch_table.copy()
+    p.dump(obj)
+    return f.getvalue()
+
+def run_python_udf_internal(pickled_python_udf):
+    try:
+        python_udf = pickle.loads(pickled_python_udf)
+        result = python_udf.func(*python_udf.args, **python_udf.kwargs)
+    except Exception as e:
+        result = "run_python_udf_internal caught exception: " + str(e)
+    return serialize(result)
+
+def load_python_udf_result_internal(pickled_python_result):
+    return pickle.loads(pickled_python_result)
+
+
+PythonUDF = collections.namedtuple("PythonUDF", ["func", "args", "kwargs"])
```

Comment:
I think we should make sure the data type sent over the wire to run an RPC is the same whether it is a torchscript or Python object. As far as I know, the torchscript pathway does not use a named tuple to transport this information. We also have to anticipate the future when Remote objects exists. What happens in the case where  we `torch.rpc(.., remote_object.foo, a, b)`?  would this still work if we are directly pickling the method of the object we are sending?

Reply:
> As far as I know, the torchscript pathway does not use a named tuple to transport this information. 
how torchscript transport this information? 

> What happens in the case where we `torch.rpc(.., remote_object.foo, a, b)`? 
remote_object will be a class instance, right? unit tests showed pickle class instance method works. Are there some caveats I did not aware of?